File: api/src/main/java/org/weso/snoicd/types/SimpleConcept.java
package org.weso.snoicd.types;


import java.util.ArrayList;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Data;

@Data
public class SimpleConcept {
	
	private String code;
	
	private List<String> descriptions;
	
	@JsonProperty("terminology_name")
	private String terminologyName;
	
	public List<String> getDescriptions() {
		if(this.descriptions == null) {
			return this.descriptions = new ArrayList<String>();
		} else {
			return this.descriptions;
		}
	}

}

File: core/src/main/java/org/weso/snoicd/core/SimpleConcept.kt
package org.weso.snoicd.core

import java.io.Serializable
import java.util.ArrayList

/**
 * A simple-concepts in icd, as a Concept is a unique code in a terminology name
 * and the set of possible descriptions. But in this case there are no related
 * concepts, it is a node with no outbound edge.
 */
data class SimpleConcept(var code: String,
                         var descriptions: List<String> = ArrayList(),
                         var terminologyName: String) : Serializable {

    companion object {

        private val serialVersionUID = 1L
    }
}
